File: MascotasUN/src/main/java/gestorAplicacion/gestionVentas/Envio.java
/*Clase que permite la gestión de la funcionalidad de los envíos programados periódicos a un cliente*/

package gestorAplicacion.gestionVentas;

import java.io.Serializable;
import java.util.ArrayList;

public class Envio implements Serializable {

	//Serializador vendedor
	private static final long serialVersionUID = 1L;
    private static ArrayList<Envio> envios = new ArrayList<>();
    
    // Atributos
    private String destino;
    private Factura factura;
    private long idEnvio;
    private String estadoEnvio = "";
    private int periodicidad;
    public enum EstadoEnvio{
        ENVIADO, CANCELADO, RECIBIDO;
    }

    //Constructor
    public Envio(long idEnvio, String destino, int periodicidad) {
        this.idEnvio= factura.getFacturaID();
        this.destino = destino;
        this.periodicidad = periodicidad;
    }
    
    //Métodos
    
    //Método que edita los atributos del envío
    public void editarEnvio(String destino, String estadoEnvio, int periodicidad) {
    	this.destino = destino;
        this.estadoEnvio = estadoEnvio;
        this.periodicidad = periodicidad;
    }
    
    //Método que cancela el envío
    public void cancelarEnvio(String estadoEnvio) {
    	this.estadoEnvio = estadoEnvio;
    }

    //Getters y setters
    
    public static ArrayList<Envio> getEnvios() {
        return envios;
    }
    
    public static void setEnvios(ArrayList<Envio> envios) {
        Envio.envios=envios;
    }
    
    public String getDestino() {
        return destino;
    }
    
    public void setDestino(String destino) {
        this.destino = destino;
    }
    
    public Factura getFactura() {
		return factura;
	}

	public void setFactura(Factura factura) {
		this.factura = factura;
	}

	public long getIdEnvio() {
		return idEnvio;
	}

	public void setIdEnvio(long idEnvio) {
		this.idEnvio = idEnvio;
	}

	public String getEstadoEnvio() {
        return estadoEnvio;
    }
	
    public void setEstadoEnvio(String estadoEnvio) {
        this.estadoEnvio = estadoEnvio;
    }
    
    public int getPeriodicidad() {
        return periodicidad;
    }
    
    public void setPeriodicidad(int periodicidad) {
        this.periodicidad = periodicidad;
    }
}

File: Practica2-MascotasUN/src/gestorAplicacion/gestionVentas/Envio.py
"""
       Clase que permite la gestión de la 
       funcionalidad de los envíos programados periódicos a un cliente
    """
    
class Envio():
    #Identificador unico de cada instancia
    idEnvio=0
    #Diccionario para guardar los envios registrados en el sistema
    envios={}
    #Constructor para la clase
    def __init__(self,destino:str,estadoEnvio:str,periodicidad:int)->None:
        Envio.idEnvio+=1
        self.destino=destino
        self.estadoEnvio=estadoEnvio
        self.periodicidad =periodicidad
        #Se guarda el objeto en un diccionario con su id como key
        Envio.envios[self.idEnvio]=self 
        #Se trae objeto de tipo factura
        self.factura=None   
    #Métodos
    
    #Método que edita los atributos del envío    
    def editarEnvio(self,destino:str,estadoEnvio:str,periodicidad:int):
        self.estadoEnvio=estadoEnvio
        self.destino=destino
        self.periodicidad=periodicidad
        
    #Metodo toString    
    def __str__(self)->str:
        return   "Identificador: "+ str(self.idEnvio) + "\n" \
            + "Destino: "+ str(self.destino) + "\n" \
            + "Estado envio: "+ str(self.estadoEnvio) + "\n" \
            + "Periodicidad: "+ str(self.periodicidad) + "\n" \
            + "Factura Asociada: "+ str(self.factura) + "\n"

    #Metodos de clase
    @classmethod
    def getEnvios(cls):
        return cls.envios
    
    @classmethod
    def setEnvios(cls, envios):
        cls.envios=envios
        
    #Getters y setters
     
    def getDestino(self):
        return self.destino
    
    def setDestino(self,destino):
        self.destino=destino
        
    def getFactura(self):
        return self.factura
        
    def setFactura(self,factura):
        self.factura=factura
        
    def getEstadoEnvio(self):
        return self.estadoEnvio
    
    def setEstadoEnvio(self,estadoEnvio:str):
        self.estadoEnvio=estadoEnvio
    
    def getPeriodicidad(self):
        return self.periodicidad
    
    def setPeriodicidad(self,periodicidad):
        self.periodicidad=periodicidad
     
        
        